hnn_core/tests/test_network.py

```diff
@@ -63,6 +63,7 @@ def test_spikes():
     spikegids = [[1, 3], [5, 7]]
     spiketypes = [['L2_pyramidal', 'L2_basket'], ['L5_pyramidal', 'L5_basket']]
     spikes = Spikes(times=spiketimes, gids=spikegids, types=spiketypes)
+    spikes.plot_input(show=False)
```

Comment:
tests could be more extensive. We could check for e.g., that errors are raised if `spike_types` are not of the right type

Reply:
I've added a [`TypeError`](https://github.com/jonescompneurolab/hnn-core/blob/b802421711cf4e8e6af3e2ad37a814575a221f65/hnn_core/viz.py#L96) check along with a matching [test](https://github.com/jonescompneurolab/hnn-core/blob/b802421711cf4e8e6af3e2ad37a814575a221f65/hnn_core/tests/test_network.py#L82). 

Do you think setting up a `ValueError` is necessary? Currently if none of the `spike_types` match the function will return a blank plot. I think this behavior makes since considering a `Spikes` object could be passed where there are no input spikes.

Reply:
see my changes. The only issue there is that spikes are not grouped nicely when `spike_types=None`, everything maps to a different color. Maybe there is a way to do it ...

Reply:
I'll leave it for now. Since we're moving towards giving users more control with the API it's probably good to be transparent about all `spike_types` present. That way they can easily define their own dictionary/modifications by just looking at the default plot.